### PR TITLE
Phaser renderer game scaling

### PR DESCRIPTION
**Pull request by Grenagar**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -576,13 +576,9 @@ const Client = IgeEventingClass.extend({
...
 		// old comment => 'on mobile increase default zoom by 25%'
-		if (ige.isMobile) {
+		/*if (ige.isMobile) {
 			zoom *= 0.75;
-		}
-
-		ige.pixi.zoom(zoom);
-		// there was a bunch of stuff involving viewports and view areas in the old method,
-		// it appeared to be out of use.
+		}*/
```
Comment (Grenagar) [inline src/client.js:581]: on mobile version zoom is increased by 25%, do you think we need it?

File: src/client.js [hunk 2]
```diff
@@ -576,13 +576,9 @@ const Client = IgeEventingClass.extend({
 	//
 	setZoom: function(zoom) {
 		// old comment => 'on mobile increase default zoom by 25%'
-		if (ige.isMobile) {
+		/*if (ige.isMobile) {
```
Comment (Grenagar) [inline src/client.js:579]: on mobile version zoom is increased by 25%, do you think we need it?

**Comment by Grenagar** [comment]
> this seems more of a hack then a fix. pretty sure camera zoom was not hardcoded initially to 0.5. what prevents you to revert initial zoom behavior to how it was before and how it worked in pixi?

I will check how it was working in pixi, in phaser-renderer btw it was not working in proper way also, so i just found some good number for visual effect.

File: ts/src/renderer/phaser/scenes/GameScene.ts [hunk 3]
```diff
@@ -209,7 +209,7 @@ class GameScene extends PhaserScene {
 
 	private setZoomSize (height: number): void {
 		// backward compatible game scaling on average 16:9 screen
```
Comment (sloont) [inline ts/src/renderer/phaser/scenes/GameScene.ts:211]: comment no longer applies

File: ts/src/renderer/phaser/scenes/GameScene.ts [hunk 4]
```diff
@@ -203,7 +205,15 @@ class GameScene extends PhaserScene {
...
 		);
-		camera.zoom = this.scale.width / 800;
+	}
+
+	private setZoomSize (height: number): void {
+		this.zoomSize = height * 2.15;
+	}
+
+	private calculateZoom(): number {
+		const scale = this.scale;
```
Comment (sloont) [inline ts/src/renderer/phaser/scenes/GameScene.ts:215]: what about
```ts
const { width, height } = this.scale;
return Math.max(width, height) / this.zoomSize;
```

File: ts/src/renderer/phaser/scenes/GameScene.ts [hunk 5]
```diff
@@ -203,7 +205,15 @@ class GameScene extends PhaserScene {
 			map.width * map.tileWidth / 2 * scaleFactor.x,
 			map.height * map.tileHeight / 2 * scaleFactor.y
 		);
-		camera.zoom = this.scale.width / 800;
+	}
+
+	private setZoomSize (height: number): void {
+		this.zoomSize = height * 2.15;
```
Comment (sloont) [inline ts/src/renderer/phaser/scenes/GameScene.ts:211]: I think we should leave a comment about this value, 2.15 being arbitrary

File: ts/src/renderer/phaser/scenes/GameScene.ts [hunk 6]
```diff
@@ -29,18 +31,18 @@ class GameScene extends PhaserScene {
...
 				gameSize, baseSize, displaySize, previousWidth, previousHeight);
 
-			camera.zoom *= gameSize.height / previousHeight;
-
-			/*camera.centerOn(
-				camera.scrollX + (gameSize.width - previousWidth) / 2,
-				camera.scrollY + (gameSize.height - previousHeight) / 2
-			);*/
+			if (this.zoomSize) {
+				camera.zoom = this.calculateZoom();
```
Comment (sloont) [inline ts/src/renderer/phaser/scenes/GameScene.ts:35]: we are directly modifying camera's zoom property here. 

on 'zoom' events we use camera.zoomTo() and a tween, but for 'resize' events we do this.

File: ts/src/renderer/phaser/scenes/GameScene.ts (hunk 3 above)
Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/scenes/GameScene.ts]: comments still applies, maybe even more. we might not be using numbers 16 and 9 explicitly but multiplying with chosen value actually accomplishes backward compatibility more accurately on 16:9 screens

File: ts/src/renderer/phaser/scenes/GameScene.ts (hunk 5 above)
Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/scenes/GameScene.ts:211]: yes, revert previous comment

File: ts/src/renderer/phaser/scenes/GameScene.ts (hunk 6 above)
Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/scenes/GameScene.ts:35]: that is correct. when we get a zoom event during game we want a smooth transition, and when window changes size or device changes orientation we want immediate change

Comment (sloont) [reply ts/src/renderer/phaser/scenes/GameScene.ts:35]: would `camera.setZoom()` be better?

Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/scenes/GameScene.ts:35]: it would be the same

**Review by sloont — APPROVED** [review]

**Pull request merged**